core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -162,6 +163,51 @@ public String defaultHostname() {
         }
     }
 
+    /**
+     * Returns the primary {@link ServerPort}'s port that this {@link Server} is listening to.
+     *
+     * @throws IllegalStateException if no available active ports
+     */
+    public int activeLocalPort() {
+        synchronized (activePorts) {
+            ServerPort serverPort = Iterables.getFirst(activePorts.values(), null);
+
+            if (serverPort == null) {
+                throw new IllegalStateException("No available active local port");
+            }
+
+            return serverPort.localAddress().getPort();
+        }
+    }
+
+    /**
+     * Returns {@link ServerPort}'s port which is mapped to the given {@code protocol}.
```

Comment:
`Returns the local {@link ServerPort} which serves the given {@link SessionProtocol}.`